### PR TITLE
protoxform: format options generically

### PR DESCRIPTION
**Pull request by lizan**
Signed-off-by: Lizan Zhou <lizan@tetrate.io>

Description:
Instead of formatting options heuristically, which will erase new annotations without changing protoxform, use proto descriptor to format options, and enforce its order as well.

Risk Level: Low
Testing: CI
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

File: tools/protoxform/protoxform.py [hunk 1]
```diff
@@ -163,24 +167,23 @@ def CamelCase(s):
   package_line = 'package %s;\n' % file_proto.package
   file_block = '\n'.join(['syntax = "proto3";\n', package_line])
 
-  options = [
-      'option java_outer_classname = "%s";' % CamelCase(os.path.basename(file_proto.name)),
-      'option java_multiple_files = true;',
-      'option java_package = "io.envoyproxy.%s";' % file_proto.package,
-  ]
+  options = file_proto.options
```
Comment (htuch) [inline tools/protoxform/protoxform.py]: Thanks for the update.

I actually consider what we today in terms of enforcing a strict set of options to be a feature not a bug. I think we should avoid having folks doing random language-specific options, which historically has resulted in a lot of inconsistency.

I do like the move away from string templates to using first class proto though.

I think we should whitelist specific options that folks want to add here, and erase others.

Comment (lizan) [reply tools/protoxform/protoxform.py]: OK, let's start from an empty FileOptions then, now it should match string template.

File: tools/protoxform/protoxform.py [hunk 2]
```diff
@@ -398,16 +376,19 @@ def FormatEnumValue(type_context, value):
...
-  if value.options.deprecated:
-    annotations.append('deprecated = true')
-  formatted_annotations = '[ %s]' % ','.join(annotations) if annotations else ''
+  formatted_annotations = FormatOptions(value.options)
   return '%s%s = %d%s;\n%s' % (leading_comment, value.name, value.number, formatted_annotations,
                                trailing_comment)
 
 
+def TextFormatValue(field, value):
+  out = io.StringIO()
```
Comment (htuch) [inline tools/protoxform/protoxform.py]: Can you add a pydoc comment explaining what this does with args/returns etc.

File: tools/protoxform/protoxform.py [hunk 3]
```diff
@@ -398,16 +376,19 @@ def FormatEnumValue(type_context, value):
...
+  text_format.PrintFieldValue(field, value, out)
+  return out.getvalue()
+
+
 def FormatOptions(options, is_message=False):
-  """Format MessageOptions/EnumOptions message.
+  """Format *Options message.
```
Comment (htuch) [inline tools/protoxform/protoxform.py]: Can you put a `e.g. MessageOptions` or whatever?

**Pull request merged**